source/server/listener_manager_impl.h

```diff
@@ -170,6 +176,7 @@ class ListenerManagerImpl : public ListenerManager, Logger::Loggable<Logger::Id:
                       WorkerFactory& worker_factory, bool enable_dispatcher_stats);
 
   void onListenerWarmed(ListenerImpl& listener);
+  void onIntelligentListenerWarmed(ListenerImpl& listener);
```

Comment:
Please remove the "intelligent" naming everywhere. Why do we need a specific function for this? Should there be a param to the existing warmed function for a warmed listener?

Reply:
Renamed to inPlaceFilterChainUpdate() for now

If I pass bool flag, there will be 3 `if` expression... Not sure which is better. Will give it a try.